frontend/src/app/document/entity-legend/entity-legend.component.html

```diff
@@ -0,0 +1,3 @@
+<div *ngFor="let entity of entities">
+    <span class="dot" [ngClass]="entity | formatEntityClass"></span><span>{{entity}}</span>
+</div>
```

Comment:
An issue with the legend specifically is that assistive technologies will parse it as a single chunk of text. The empty `<span>` elements will be ignored in the accessibility tree, so the inner structure will appear the same as:

```html
<div>
    Person Location Miscellaneous Organization
</div>
```

So this element is missing some meaningful information about its role or internal structure, including how each category relates to the content.

In this case, the element could use some semantic tags. For example, it could be marked up like this:

```html
<footer>
    <p>Legend</p>
    <ul>
        <li>Person</li>
        <li>Location</li>
        <li>Miscellaneous</li>
        <li>Organization</li>
    </ul>
</footer>
```

(or use [`<dl>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl) if you're feeling fancy).

That's still missing information about how "Person" actually relates to the content, though, but that depends on how the highlights are implemented.